pkg/controller/syncsetinstance/syncsetinstance_controller.go

```diff
@@ -302,6 +325,25 @@ func (r *ReconcileSyncSetInstance) Reconcile(request reconcile.Request) (reconci
 	applier := r.applierBuilder(restConfig, ssiLog)
 	applyErr := r.applySyncSet(ssi, spec, dynamicClient, applier, ssiLog)
 	ssi.Status.Applied = applyErr == nil
+	if ssi.Status.Applied && ssi.Status.FirstSuccessTimestamp == nil {
+		now := metav1.Now()
+		ssi.Status.FirstSuccessTimestamp = &now
+		startTime := ssi.CreationTimestamp.Time
+		if cd.Status.InstalledTimestamp != nil && startTime.Before(cd.Status.InstalledTimestamp.Time) {
+			startTime = cd.Status.InstalledTimestamp.Time
+		}
+		if existingCondition := controllerutils.FindClusterDeploymentCondition(cd.Status.Conditions,
+			hivev1.UnreachableCondition); existingCondition != nil {
+			if existingCondition.LastTransitionTime.Time.After(startTime) {
+				startTime = existingCondition.LastTransitionTime.Time
+			}
+		}
+		applyTime := ssi.Status.FirstSuccessTimestamp.Sub(startTime).Seconds()
+		metricTimeToApplySyncset.Observe(float64(applyTime))
+		if sss := ssi.Spec.SelectorSyncSetRef; sss != nil {
+			metricTimeToApplySelectorSyncset.WithLabelValues(sss.Name).Observe(float64(applyTime))
+		}
+	}
```

Comment:
I'm just realizing what will happen when this change rolls out for all existing clusters, we'll get a bunch of first success showing weeks and months. I think they keep a pretty limited window of time, 2 weeks iirc, so in theory this will trend down quickly as new clusters are added and then be fine after a couple weeks. This seems ok to me I think, I can't think of an option other than dont't report metrics over 24h or something.

Reply:
Could we check for `ssi.Status.Applied != original.Status.Applied` as well?

Reply:
Or just `!original.Status.Applied`?

Reply:
And if `original.Status.Applied` is true, then set `ssi.Status.FirstSuccessTimestamp` but do not observe the metric.

Reply:
This sounds pretty good to me, what do you think @suhanime

Reply:
Yes it seems to cover all scenarios.